src/ReturnTypes/CollectionGenericStaticMethodDynamicMethodReturnTypeExtension.php

```diff
@@ -123,7 +123,9 @@ private function handleGenericObjectType(ClassReflection $classReflection, Class
                 if ($type instanceof GenericObjectType && (($innerTypeReflection = $type->getClassReflection()) !== null)) {
                     $genericTypes = $innerTypeReflection->typeMapToList($innerTypeReflection->getActiveTemplateTypeMap());
 
-                    return new GenericObjectType($classReflection->getName(), $genericTypes);
+                    if ($classReflection->isSubclassOf($type->getClassName())) {
```

Comment:
What happens if this condition is not met? And in which cases we would hit the `else` here? 🤔

Reply:
If the `$classReflection` is not a subtype of `Collection` we will just traverse through the types and not modify them. This is what is (correctly) happening in the test cases that I added where it traverses through `Ok` => `Some` and keeps them as they are as they are not a subtype of `Collection`.